### DISCUSSION TITLE
Feature request: Relative date options

### DISCUSSION THREAD

**Discussion by Laskewitz** [microsoft/Power-Fx#1989] [General]
I'm not sure if this is the place for this, but I don't know where else to put it!

I've been a fan of [Moment.js](https://momentjs.com/) for years. It's a library that you can use for various date calculations/formatting in different languages. Now, Power Fx handles calculations & formatting pretty well. There's only one thing missing to me: relative dates. I would love Power Fx to have that as well (without a lot of formula usage...).

I would love to see some formula or extra formula where we can input a date and then it will calculate how long that's from the current date: 1 hour ago, 2 days ago, in 2 days, in 2 hours, etc.

Is this an option for Power Fx?

**Comment by russgove** [comment]
hi,
does this meet you needs?
https://docs.microsoft.com/en-us/powerapps/maker/canvas-apps/functions/function-dateadd-datediff
Russell

**Comment by Laskewitz** [comment]
Hi @russgove,

It does meet it partly, the difference with moment.js is that you don't have to think about if it's minutes, hours, days, or months. That will go automatically. So it saves a lot of if statements.

Daniel

**Comment by anderson-joyle** [comment]
This was an issue originally. It was closed and converted into a discussion.
